test/quoting_ir.jl

```diff
@@ -42,3 +42,57 @@ quote
     $$x + 1
 #    └┘ ── `$` expression outside string or quote block
 end
+
+########################################
+# Quoted property access with identifier
+Core.:(foo)
+#---------------------
+1   TestMod.Core
+2   (call top.getproperty %₁ :foo)
+3   (return %₂)
+
+########################################
+# Quoted property access with operator
+Core.:(!==)
+#---------------------
+1   TestMod.Core
+2   (call top.getproperty %₁ :!==)
+3   (return %₂)
+
+########################################
+# Quoted property access on variable
+let
+    x = (a=1, b=2)
+    x.:(a)
+end
+#---------------------
+1   (call core.tuple :a :b)
+2   (call core.apply_type core.NamedTuple %₁)
+3   (call core.tuple 1 2)
+4   (= slot₁/x (call %₂ %₃))
+5   slot₁/x
+6   (call top.getproperty %₅ :a)
+7   (return %₆)
+
+########################################
+# Quoted operator function definition (issue #20)
+function Base.:(==)(a::Issue20, b::Issue20)
+    a === b
+end
```

Comment:
As we don't actually execute these IR cases we're only looking for structural equivalence of the simplest possible code in the desugaring of interest. So I'd suggest simplifying:

```suggestion
function Base.:(==)()
end
```

(Possibly we don't even need this case because it does end up going via the same structural simplification code path as other getproperty lowering?  And it's covered nicely by the tests in test/quoting.jl?)